### PR TITLE
Upgrade three.js to 0.106.2

### PR DESCRIPTION
**Pull request by BrendanAnnable**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +71,6 @@
     "@types/socket.io-client": "^1.4.29",
     "@types/storybook__addon-actions": "^3.4.3",
     "@types/storybook__react": "^4.0.2",
-    "@types/three": "^0.91.0",
```
Comment (BrendanAnnable) [inline package.json]: Types are now built in!

File: src/client/components/localisation/utils.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { createTransformer } from 'mobx-utils'
 import { MeshLambertMaterial } from 'three'
 import { Color } from 'three'
-import { JSONLoader } from 'three'
 import { Geometry } from 'three'
 import { Material } from 'three'
+import { LegacyJSONLoader } from 'three/examples/jsm/loaders/deprecated/LegacyJSONLoader'
```
Comment (BrendanAnnable) [inline src/client/components/localisation/utils.ts]: Yay we can import from three.js examples now 😄

File: src/client/image_decoder/image_decoder.ts [hunk 3]
```diff
@@ -153,7 +153,8 @@ export class ImageDecoder {
       scene.remove(...this.bayerDecoder!.scene.children)
       scene.add(mesh)
 
-      this.renderer.render(scene, this.bayerDecoder!.camera, target)
+      this.renderer.setRenderTarget(target)
+      this.renderer.render(scene, this.bayerDecoder!.camera)
```
Comment (BrendanAnnable) [inline src/client/image_decoder/image_decoder.ts]: The render method no longer takes a render target

File: tslint.json [hunk 4]
```diff
@@ -53,6 +53,7 @@
     "no-reference-import": true,
     "no-require-imports": true,
     "no-shadowed-variable": false,
+    "no-submodule-imports": [true, "three/examples/jsm"],
```
Comment (BrendanAnnable) [inline tslint.json]: Submodule imports are banned by default, so this adds threejs examples to the whitelist of exceptions.

File: src/client/components/three/builders.ts [hunk 5]
```diff
@@ -152,8 +152,7 @@ export const shaderMaterial = createUpdatableComputed(
 )
 
 type TypedArray
-  = ArrayBuffer
```
Comment (BrendanAnnable) [inline src/client/components/three/builders.ts]: `ArrayBuffer` is no longer allowed

File: src/client/components/localisation/utils.ts [hunk 6]
```diff
@@ -33,7 +33,7 @@ const coloredMaterial = (material: Material, color?: string) => {
 }
 
 const parseConfig = createTransformer((config: any) => {
-  const { geometry, materials } = new JSONLoader().parse(config)
+  const { geometry, materials } = new LegacyJSONLoader().parse(config, '/')
```
Comment (BrendanAnnable) [inline src/client/components/localisation/utils.ts]: The types require a `path` to be given, but we have no external resources to load anyway. So it is not actually used.

File: package.json (hunk 1 above)
Comment (TrentHouliston) [reply package.json]: Hurray!

File: tslint.json (hunk 4 above)
Comment (TrentHouliston) [reply tslint.json]: Hopefully, we won't need to use examples anymore

File: src/client/components/vision/camera/view.tsx [hunk 7]
```diff
@@ -72,6 +72,7 @@ export class CameraView extends Component<{ viewModel: CameraViewModel }> {
     const { viewModel } = this.props
     const renderer = viewModel.renderer(viewModel.canvas)
     if (renderer) {
+      renderer.setRenderTarget(null)
```
Comment (BrendanAnnable) [inline src/client/components/vision/camera/view.tsx]: I'm actually surprised it worked without this, but I believe this is the correct thing to do.

(and all goes away with my refactor)

**Pull request merged**